cloud_automation/gcf/bt_trigger.go

```diff
@@ -20,31 +20,94 @@ import (
 
 	"cloud.google.com/go/bigtable"
 	"cloud.google.com/go/storage"
+	"google.golang.org/api/dataflow/v1b3"
 	"google.golang.org/grpc/codes"
 	"google.golang.org/grpc/status"
 )
 
 const (
-	projectID          = "google.com:datcom-store-dev"
-	bigtableInstance   = "prophet-cache"
-	bigtableCluster    = "prophet-cache-c1"
+	PROD							= "prod"
+	TEST							= "test"
+
+	btProjectID				= "google.com:datcom-store-dev"
+	dataBucket				= "prophet_cache"
+	dataFilePattern		= "cache.csv*"
+	dataflowTemplate	= "gs://datcom-dataflow-templates/templates/csv_to_bt_improved"
 	createTableRetries = 3
-	columnFamily       = "csv"
-	bigtableNodesHigh  = 300
-	bigtableNodesLow   = 20
-	triggerFile        = "latest_base_cache_version.txt"
-	successFile        = "success.txt"
-	failureFile        = "failure.txt"
+	columnFamily			= "csv"
+	baseCacheType			= "base"
+	branchCacheType		= "branch"
+
+	// NOTE: The following three files represents the state of a BT import. They
+	// get written under:
+	//
+	//		gs://<env.ControlBucket>/(base|branch)/<TableID>/
+	//
+	// Init: written by borg to start BT import.
+	initFile					 = "init.txt"
+	// Launched: written by this cloud function to mark launching of BT import job.
+	launchedFile			 = "launched.txt"
+	// Completed: written by dataflow to mark completion of BT import.
+	completedFile			= "completed.txt"
+
+	// TODO: Deprecate these files.
+	triggerFile				= "latest_base_cache_version.txt"
+	successFile				= "success.txt"
+	failureFile				= "failure.txt"
 	airflowTriggerFile = "airflow_trigger.txt"
 )
 
+type Environment struct {
+	// BT instance holding base caches.
+	BaseBTInstance string
```

Comment:
nit: Capital names means it's exported and used by other modules, if it's internal can just be lower case